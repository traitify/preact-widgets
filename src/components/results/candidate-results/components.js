export CandidateResults from "./index";
export PersonalityArchetype from "./personality-archetype";
export PersonalityDetails from "./personality-details";
export PersonalityDimensionColumn from "./personality-dimension-column";
export PersonalityDimensionColumns from "./personality-dimension-columns";
export PersonalityDimensionDetails from "./personality-dimension-details";
export PersonalityDimensions from "./personality-dimensions";
