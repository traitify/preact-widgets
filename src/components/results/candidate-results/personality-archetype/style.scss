@import "~style/helpers";

.container {
  @extend %widgetContainer;
  @extend %cf;
}
.details {
  background: #4d53d6;
  background: -moz-linear-gradient(137deg, #4d85d6 0%, #4d53d6 100%);
  background: -webkit-linear-gradient(137deg,#4d85d6 0%,#4d53d6 100%);
  background: linear-gradient(137deg, #4d85d6 0%,#4d53d6 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#4d85d6', endColorstr='#4d53d6',GradientType=1 );
  color: $white;
  border-radius: 5px;
  padding: 20px;
  margin-bottom: 20px;
  @include rems(16);
  @include mq(m) {
    float: left;
    width: 49%;
    margin-right: 2%;
    padding: 20px;
    @include rems(17);
  }
  @include mq(xl) {
    @include rems(18);
  }
  img {
    float: left;
    margin-right: 20px;
    margin-bottom: 20px;
    max-width: 80px;
  }
  h2 {
    margin-top: 5px;
  }
  span {
    font-weight: bold;
    display: block;
  }
  p {
    clear: both;
  }
}
.meaning {
  padding: 20px;
  margin-bottom: 20px;
  @include rems(16);
  @include mq(m) {
    float: left;
    width: 49%;
    @include rems(17);
  }
  @include mq(xl) {
    @include rems(18);
  }
  p {
    margin: 0;
  }
}
.thirdPersonDetails {
  @extend .details;
  width: 100%;
  display: flex;
  flex-direction: column;
  @include mq(l) {
    display: inline;
    .itemTwo {
      width: 50%;
      padding-left: 3%;
      float: right;
      padding-top: 30px;
    }
    .itemOne {
      width: 45%;
      display: inline-block;
      border-right: 2px solid $white;
    }
    .deckAndHeading {
      width: 100%;
      padding-right: 5%;
    }
    span {
      display: inline;
      width: 50%;
    }
    .personalityHeading {
      padding-top: 20px;
    }
  }
  img {
    height: 130px;
    max-width: 130px;
  }
  h2 {
    margin-bottom: 0;
    margin-right: 0;
  }
}
