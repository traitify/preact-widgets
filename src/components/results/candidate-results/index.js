import {
  faThLarge,
  faUser
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import Icon from "lib/helpers/icon";
import withTraitify from "lib/with-traitify";
import Guide from "components/results/guide";
import PersonalityArchetype from "components/results/personality/archetype/heading";
import PersonalityDimensions from "components/results/personality/dimension/list";
import PersonalityTips from "components/results/personality/archetype/tips";
import PersonalityTraits from "components/results/personality/trait/list";
import style from "../style.scss";

function CandidateResults(props) {
  const {getOption, translate} = props;
  let disabledComponents = getOption("disabledComponents") || [];
  const allowHeaders = getOption("allowHeaders");
  const thirdPerson = getOption("perspective") === "thirdPerson";
  if(!thirdPerson) { disabledComponents = [...disabledComponents, "PersonalityPitfalls"]; }

  return (
    <section>
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faUser} /> {translate("personality_type")}</p>
      )}
      <PersonalityArchetype {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_details")}</p>
      )}
      <PersonalityTips {...props} />
      {allowHeaders && (
        <p className={style.lessMarginBottom}><Icon icon={faThLarge} /> {translate("personality_advice")}</p>
      )}
      <PersonalityDimensions {...props} disabledComponents={disabledComponents} />
      {thirdPerson && (
        <div>
          {allowHeaders && (
            <p className={style.heading}><Icon icon={faThLarge} /> {translate("interview_guide_heading")}</p>
          )}
          <Guide {...props} />
        </div>
      )}
      {allowHeaders && (
        <p className={style.paddingTop}><Icon icon={faThLarge} /> {translate("personality_traits")}</p>
      )}
      <PersonalityTraits {...props} />
    </section>
  );
}

CandidateResults.propTypes = {
  getOption: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default withTraitify(CandidateResults);
