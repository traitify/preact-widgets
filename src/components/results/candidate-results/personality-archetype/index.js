import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityArchetype extends Component {
  static defaultProps = {assessment: null, deck: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    deck: PropTypes.shape({name: PropTypes.string.isRequired}),
    followDeck: PropTypes.func.isRequired,
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    options: PropTypes.shape({archetype: PropTypes.object}).isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityArchetype.initialized", this);
    this.props.followDeck();
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityArchetype.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }
    if(!this.props.isReady("deck")) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    if(disabledComponents.includes("PersonalityArchetype")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }
    const badge = personality.details.find(({title}) => title === "Badge");
    const perspective = this.props.getOption("perspective");
    const description = (
      perspective === "thirdPerson"
        ? personality.details.find(({title}) => title === "Hiring Manager Description")
        : personality.details.find(({title}) => title === "Candidate Description")
    );
    const splitIfPresent = description && style.badgeAndName;

    return (
      <div className={style.container}>
        {perspective === "thirdPerson" ? (
          <div className={style.thirdPersonDetails}>
            <div className={splitIfPresent}>
              {badge && <img alt={personality.name} src={badge.body} />}
              <DangerousHTML
                html={this.props.translate("personality_heading_third_person", {
                  deck_name: this.props.deck.name,
                  personality: `<br /><span>${personality.name}</span><br />`
                })}
                className={style.personalityHeading}
                tag="h2"
              />
            </div>
            {description && <span className={style.divider} />}
            {description && <span className={style.body}>{description.body}</span>}
          </div>
        ) : [
          <div key="heading" className={style.details}>
            {badge && <img alt={personality.name} src={badge.body} />}
            <DangerousHTML
              html={this.props.translate("personality_heading", {
                deck_name: this.props.deck.name,
                personality: `<span>${personality.name}</span>`
              })}
              tag="h2"
            />
            {description && <p>{description.body}</p>}
          </div>,
          <div key="meaning" className={style.meaning}>
            <DangerousHTML html={this.props.translate("candidate_description_for_archetype_html")} tag="p" />
          </div>
        ]}
      </div>
    );
  }
}

export {PersonalityArchetype as Component};
export default withTraitify(PersonalityArchetype);
