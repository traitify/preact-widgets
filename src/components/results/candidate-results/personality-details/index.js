import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDetails.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const personality = this.props.assessment.archetype || {};
    const {details} = personality;
    if(!details) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    let disableSettings = disabledComponents.includes("PersonalitySettings");
    let disableTools = disabledComponents.includes("PersonalityTools");
    const settings = details.filter(({title}) => (title === "Settings that Work for You")).map(({body}) => body);
    const tools = details.filter(({title}) => (title === "Tools to Use")).map(({body}) => body);
    if(settings.length === 0) { disableSettings = true; }
    if(tools.length === 0) { disableTools = true; }
    if(disableSettings && disableTools) { return null; }

    return (
      <div className={style.container}>
        {!disableTools && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#008dc7"}} />
              <h4 className={style.title} style={{color: "#008dc7"}}>{this.props.translate("candidate_heading_for_tools")}</h4>
              <ul className={style.description}>
                {tools.map((tool) => (
                  <li key={tool}>{tool}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        {!disableSettings && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#32be4b"}} />
              <h4 className={style.title} style={{color: "#32be4b"}}>{this.props.translate("candidate_heading_for_settings")}</h4>
              <ul className={style.description}>
                {settings.map((setting) => (
                  <li key={setting}>{setting}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        <div className={style.clearfix} />
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
