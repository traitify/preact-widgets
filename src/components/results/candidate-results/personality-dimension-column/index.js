import PropTypes from "prop-types";
import {Component} from "react";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityDimensionColumn extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumn.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumn.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, level, name}, score}} = this.props;
    const color = `#${badge.color_1}`;
    let backgroundClass;
    switch(true) {
      case score <= 3: backgroundClass = "low"; break;
      case score <= 6: backgroundClass = "medium"; break;
      default: backgroundClass = "high";
    }

    return (
      <li className={style.container}>
        <div>
          <div className={style.icon} style={{border: `3px solid ${color}`, background: rgba(color, 8.5)}}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </div>
          <h3>{name}</h3>
          <h2>{level}</h2>
        </div>
        <div className={`${style.background} ${style[backgroundClass]}`} style={{background: rgba(color, 10)}} />
      </li>
    );
  }
}

export {PersonalityDimensionColumn as Component};
export default withTraitify(PersonalityDimensionColumn);
