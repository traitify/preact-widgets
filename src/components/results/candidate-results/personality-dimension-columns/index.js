import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityDimensionColumn from "../personality-dimension-column";
import style from "./style";

class PersonalityDimensionColumns extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({personality_types: PropTypes.array}),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumns.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumns.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const types = this.props.assessment.personality_types.sort((x, y) => {
      const xDetail = x.personality_type.details.find(({title}) => title === "Position") || {};
      const yDetail = y.personality_type.details.find(({title}) => title === "Position") || {};

      return (xDetail.body || 1) - (yDetail.body || 1);
    });

    return (
      <div className={style.container}>
        <p>{this.props.translate("candidate_description_for_dimensions")}</p>
        <ul className={style.columns}>
          {types.map((type) => (
            <PersonalityDimensionColumn
              key={type.personality_type.id}
              type={type}
              {...this.props}
            />
          ))}
        </ul>
        <div className={style.columnsBuffer} />
      </div>
    );
  }
}

export {PersonalityDimensionColumns as Component};
export default withTraitify(PersonalityDimensionColumns);
