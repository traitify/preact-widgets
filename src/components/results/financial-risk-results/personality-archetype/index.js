import PropTypes from "prop-types";
import {Component} from "react";
import {detailWithPerspective} from "lib/helpers";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityArchetype extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityArchetype.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityArchetype.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }
    const badge = personality.details.find(({title}) => title === "Badge").body;
    const perspective = this.props.getOption("perspective");
    const options = {base: personality, perspective};
    const pronoun = perspective === "thirdPerson" ? "Their" : "Your";

    return (
      <div className={style.archetype}>
        <div className={style.archetypeStyle}>
          <img alt={personality.name} src={badge} />
          <h2>{pronoun} Financial Risk style is <span>{personality.name}</span></h2>
          <p>{detailWithPerspective({...options, name: "Description"})}</p>
        </div>
        <div className={style.archetypeMeaning}>
          <h2>What does this tell me?</h2>
          <p>{detailWithPerspective({...options, name: "Analysis"})}</p>
        </div>
      </div>
    );
  }
}

export {PersonalityArchetype as Component};
export default withTraitify(PersonalityArchetype);
