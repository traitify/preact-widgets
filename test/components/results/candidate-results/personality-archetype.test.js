import {Component} from "components/results/candidate-results/personality-archetype";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";
import deck from "support/json/deck/big-five.json";

jest.mock("lib/with-traitify", () => ((value) => value));

const assessmentWithDetails = {
  ...assessment,
  archetype: {
    ...assessment.archetype,
    details: [
      ...assessment.archetype.details,
      {body: "https://cdn.traitify.com/frtq/conservative_white.png", title: "Badge"},
      {body: "Maybe just hire qualified candidats", title: "Hiring Manager Description"}
    ]
  }
};

describe("PersonalityArchetype", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      deck,
      followDeck: jest.fn().mockName("followDeck"),
      getOption: jest.fn().mockName("getOption"),
      isReady: jest.fn().mockName("isReady").mockReturnValue(true),
      options: {},
      translate: jest.fn().mockName("translate").mockImplementation((value, options = {}) => `${value}, ${options}`),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityArchetype.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityArchetype.updated", component.instance);
    });
  });

  it("follows the deck", () => {
    new ComponentHandler(<Component {...props} />);

    expect(props.followDeck).toHaveBeenCalled();
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component in third person", () => {
    props.getOption.mockImplementation((key) => (key === "perspective" ? "thirdPerson" : []));
    props.assessment = assessmentWithDetails;
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with badge", () => {
    props.assessment = assessmentWithDetails;
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", () => {
    props.getOption.mockReturnValue(["PersonalityArchetype"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if deck not ready", () => {
    props.deck = null;
    props.isReady.mockImplementation((value) => value !== "deck");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no archetype", () => {
    props.assessment = {...props.assessment, archetype: null};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
