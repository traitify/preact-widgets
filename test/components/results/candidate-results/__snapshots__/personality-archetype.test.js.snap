// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityArchetype renders component 1`] = `
<div>
  <div>
    <h2
      dangerouslySetInnerHTML={
        Object {
          "__html": "personality_heading, [object Object]",
        }
      }
    />
    <p>
      You can get lost in your work, appreciating the value of devoting expertise to the execution of a task. You are at ease working in your own way, on your own terms.
    </p>
  </div>
  <div>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "candidate_description_for_archetype_html, [object Object]",
        }
      }
    />
  </div>
</div>
`;

exports[`PersonalityArchetype renders component in third person 1`] = `
<div>
  <div>
    <div>
      <img
        alt="Cohort"
        src="https://cdn.traitify.com/frtq/conservative_white.png"
      />
      <div>
        <h2
          dangerouslySetInnerHTML={
            Object {
              "__html": "personality_heading, [object Object]",
            }
          }
        />
      </div>
    </div>
    <span>
      Maybe just hire qualified candidats
    </span>
  </div>
</div>
`;

exports[`PersonalityArchetype renders component with badge 1`] = `
<div>
  <div>
    <img
      alt="Cohort"
      src="https://cdn.traitify.com/frtq/conservative_white.png"
    />
    <h2
      dangerouslySetInnerHTML={
        Object {
          "__html": "personality_heading, [object Object]",
        }
      }
    />
    <p>
      You can get lost in your work, appreciating the value of devoting expertise to the execution of a task. You are at ease working in your own way, on your own terms.
    </p>
  </div>
  <div>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "candidate_description_for_archetype_html, [object Object]",
        }
      }
    />
  </div>
</div>
`;

exports[`PersonalityArchetype renders nothing if deck not ready 1`] = `null`;

exports[`PersonalityArchetype renders nothing if disabled 1`] = `null`;

exports[`PersonalityArchetype renders nothing if no archetype 1`] = `null`;

exports[`PersonalityArchetype renders nothing if results not ready 1`] = `null`;
